src/App.tsx

```diff
@@ -1,29 +1,55 @@
-import React from 'react';
+/* eslint-disable @typescript-eslint/indent */
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42).map(n => `Item ${n}`);
+const TOTAL_ITEMS = 42;
+const items = getNumbers(1, TOTAL_ITEMS).map(n => `Item ${n}`);
+const itemsPerPage = [3, 5, 10, 20];
 
 export const App: React.FC = () => {
+  const [perPage, setPerPage] = useState(itemsPerPage[1]);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const itemsFrom = perPage * currentPage - perPage + 1;
+  const itemsTo = Math.min(perPage * currentPage, TOTAL_ITEMS);
+
+  const visibleItems = items.slice(itemsFrom - 1, itemsTo);
```

Comment:
Use the function - `getNumbers` immediately. 
```suggestion
  const visibleItems = getNumbers(itemsFrom  -1 , itemsTo));
  const listItems = visibleItems.map(n => `Item ${n}`;
```